### PR TITLE
FIX FLINK-23580 

### PR DESCRIPTION
**Pull request by shutter-cp**
add clinkHouse JDBC

### CODE REVIEW

File: flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/ClinkHouseDialect.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import org.apache.flink.table.types.logical.LogicalTypeRoot;
+import org.apache.flink.table.types.logical.RowType;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/** JDBC dialect for MySQL. */
+public class ClinkHouseDialect extends AbstractDialect {
```
Comment (paul8263) [inline flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/ClinkHouseDialect.java:32]: Please correct the typo. It should be ClickHouse. Also the comments are for MySQL. Please correct them.

File: flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/converter/ClinkHouseRowConverter.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+import java.math.BigInteger;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Runtime converter that responsible to convert between JDBC object and Flink internal object for
+ * ClinkHouse.
+ */
+public class ClinkHouseRowConverter extends AbstractJdbcRowConverter {
```
Comment (paul8263) [inline flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/converter/ClinkHouseRowConverter.java:33]: Also the Typo here.

File: flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/converter/ClinkHouseRowConverter.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public String converterName() {
+        return "ClinkHouseSQL";
```
Comment (paul8263) [inline flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/converter/ClinkHouseRowConverter.java:39]: Should be ClickHouse

File: flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/converter/ClinkHouseRowConverter.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public ClinkHouseRowConverter(RowType rowType) {
```
Comment (paul8263) [inline flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/converter/ClinkHouseRowConverter.java:42]: Should be ClickHouse

File: flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/JdbcDialects.java [hunk 5]
```diff
@@ -26,7 +26,11 @@
 public final class JdbcDialects {
 
     private static final List<JdbcDialect> DIALECTS =
-            Arrays.asList(new DerbyDialect(), new MySQLDialect(), new PostgresDialect());
+            Arrays.asList(
+                    new DerbyDialect(),
+                    new MySQLDialect(),
+                    new PostgresDialect(),
+                    new ClinkHouseDialect());
```
Comment (paul8263) [inline flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/JdbcDialects.java:33]: Should be ClickHouse

**Review by paul8263 — CHANGES_REQUESTED** [review]

**Review by shutter-cp — COMMENTED** [review]
Changes completed

File: flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/ClickHouseDialect.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+    @Override
+    public Optional<String> defaultDriverName() {
+        return Optional.of("ru.yandex.clickhouse.ClickHouseDriver");
+    }
+
+    @Override
+    public String quoteIdentifier(String identifier) {
+        if (identifier.contains(".")) {
+            int dotIndex = identifier.indexOf(".");
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/ClickHouseDialect.java:66]: In some degenerated cases table and column names could contain dots and... probably in this case this logic will not work e.g.
```sql
CREATE TABLE my_db_name.`My.Table`
(

    `My.Column` String
)
ENGINE = MergeTree
ORDER BY `My.Column`
SETTINGS index_granularity = 8192;
```

File: flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/ClickHouseDialect.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        return MAX_TIMESTAMP_PRECISION;
+    }
+
+    @Override
+    public int minTimestampPrecision() {
+        return MIN_TIMESTAMP_PRECISION;
+    }
+
+    @Override
+    public List<LogicalTypeRoot> unsupportedTypes() {
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/ClickHouseDialect.java:136]: I fear this should be reworked as https://issues.apache.org/jira/browse/FLINK-24253 introduced some major changes here ...

**Comment by afedulov** [comment]
Hi @shutter-cp , thank you for the contribution. Please fill out the provided PR template to share the required description of the contribution and refer to https://flink.apache.org/contributing/code-style-and-quality-pull-requests.html for the PR naming conventions.

**Comment by MartijnVisser** [comment]
@shutter-cp We've now moved the code from the JDBC connector to its own dedicated repository; please re-route this PR to https://github.com/apache/flink-connector-jdbc

**Pull request closed without merging**